c2rust-transpile/src/lib.rs

```diff
@@ -227,10 +227,65 @@ fn get_module_name(
 
 /// Main entry point to transpiler. Called from CLI tools with the result of
 /// clap::App::get_matches().
-pub fn transpile(tcfg: TranspilerConfig, cc_db: &Path, extra_clang_args: &[&str]) {
+pub fn transpile(tcfg: TranspilerConfig, source_or_cc_db: &Path, extra_clang_args: &[&str]) {
     diagnostics::init(tcfg.enabled_warnings.clone(), tcfg.log_level);
 
-    let lcmds = get_compile_commands(cc_db, &tcfg.filter).unwrap_or_else(|_| {
+    let build_dir = get_build_dir(&tcfg, source_or_cc_db);
+    let mut temp_json_path = std::env::temp_dir();
+    temp_json_path.push("compile_commands.json");
+
+    let ext = source_or_cc_db.extension().unwrap_or_default();
+    let is_c_or_h_file = ext == "c" || ext == "h";
+    let is_wildcard = source_or_cc_db
+        .file_name()
+        .unwrap_or_default()
+        .to_string_lossy()
+        .contains('*');
+
+    let format_compile_command = |build_dir: &Path, file_path: &Path| -> String {
+        format!(
+            r#"{{
+                "directory": "{}",
+                "command": "clang {}",
+                "file": "{}"
+            }}"#,
+            build_dir.to_str().unwrap(),
+            file_path.to_str().unwrap(),
+            file_path.to_str().unwrap()
+        )
+    };
```

Comment:
I think a `#[derive(Serialize)]` on a `struct CompileCommand` would be a lot simpler to use and less error-prone. That way we can leave things as `PathBuf`s, etc.

Also, an `arguments` list field instead of a `command` field is usually what automated tooling seems to generate, as spacing isn't a concern there.

Reply:
Oh, we already have a `#[derive(Serialize, Deserialize)] struct CompileCmd`: https://github.com/immunant/c2rust/blob/c603b25b517303b2f7fbf81bf00d1daac896e364/c2rust-transpile/src/compile_cmds/mod.rs#L12-L33